src/main/java/dao_layer/service/FamilyService.java

```diff
@@ -1,104 +1,135 @@
 package dao_layer.service;
 
-import dao_layer.dao.FamilyDao;
+import dao_layer.dao.CollectionFamilyDao;
 import happy_family.Family;
 import happy_family.Human;
 import happy_family.Pet;
 
-import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 import java.util.Set;
-import java.util.stream.IntStream;
 
 public class FamilyService {
+    private CollectionFamilyDao familyDao;
 
-    private final FamilyDao familyDao;
-    private List<Family> families = new ArrayList<>();
 
-
-    public FamilyService(FamilyDao familyDao) {
-        this.familyDao = familyDao;
-        this.families = familyDao.getAllFamilies();
+    public FamilyService() {
+        this.familyDao = new CollectionFamilyDao();
     }
 
+
     public void fillWithTestData() {
-        Human father1 = new Human("John", "Doe", 1980L);
-        Human mother1 = new Human("Jane", "Doe", 1985L);
-        Family family1 = new Family(mother1, father1);
+        Family[] families = {
+                new Family(new Human("Emma", "Smith", 1985L), new Human("John", "Smith", 1980L)),
+                new Family(new Human("Olivia", "Johnson", 1988L), new Human("James", "Johnson", 1985L)),
+                new Family(new Human("Sophia", "Williams", 1990L), new Human("Liam", "Williams", 1989L))
+        };
 
-        Human father2 = new Human("Jim", "Beam", 1975L);
-        Human mother2 = new Human("Jill", "Beam", 1980L);
-        Family family2 = new Family(mother2, father2);
+        families[0].addChild(new Human("Alice", "Smith", 2010L));
+        families[0].addChild(new Human("Bob", "Smith", 2012L));
+        families[1].addChild(new Human("Charlie", "Johnson", 2015L));
+        families[2].addChild(new Human("Mia", "Williams", 2018L));
 
-        families.add(family1);
-        families.add(family2);
-    }
+        for (Family family : families) {
```

Comment:
also you can implement` saveAllFamilies(List<Family> families)` in DAO side for this purpose